xarray/tutorial.py

```diff
@@ -6,23 +6,78 @@
 
 """
 import hashlib
-import os as _os
-from urllib.request import urlretrieve
+import pathlib
+import shutil
+import tempfile
 
 import numpy as np
+import requests
 
 from .backends.api import open_dataset as _open_dataset
+from .backends.rasterio_ import open_rasterio as _open_rasterio
 from .core.dataarray import DataArray
 from .core.dataset import Dataset
 
-_default_cache_dir = _os.sep.join(("~", ".xarray_tutorial_data"))
+_cache_name = "xarray_tutorial_data"
+_cache_dir = pathlib.Path.home() / ".cache"
+# TODO: I/O on import. Might not be a good idea.
+if _cache_dir.exists():
+    _default_cache_dir = _cache_dir / _cache_name
+else:
+    _default_cache_dir = pathlib.Path.home() / f".{_cache_name}"
 
 
-def file_md5_checksum(fname):
-    hash_md5 = hashlib.md5()
-    with open(fname, "rb") as f:
-        hash_md5.update(f.read())
-    return hash_md5.hexdigest()
+def check_md5sum(content, checksum):
+    md5 = hashlib.md5()
+    md5.update(content)
+    md5sum = md5.hexdigest()
+
+    return md5sum == checksum
+
+
+def download_to(url, path):
+    with requests.get(url, stream=True) as r, path.open("wb") as f:
```

Comment:
nit: I usually find it a little easier to read when you nest context managers:
```suggestion
    with requests.get(url, stream=True) as r
        with path.open("wb") as f:
            ...
```